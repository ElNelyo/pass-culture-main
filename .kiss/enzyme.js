// ./node_modules/.bin/jest --env=jsdom ./path/to/file.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = { errors: {} }
      // when
      const wrapper = shallow(<MyComponent {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
