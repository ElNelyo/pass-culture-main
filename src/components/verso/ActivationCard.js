/* eslint
  react/jsx-one-expression-per-line: 0 */
import React from 'react'
//
import ActivationButton from '../layout/buttons/ActivationButton'

const ActivationCard = () => (
  <div
    id="verso-tuto-activation"
    className="flex-rows is-full-height is-white-text text-center"
  >
    <p className="flex-0 text-center is-italic fs20">
      <span className="is-block">Et voil√† !</span>
      <span className="is-block">vous pouvez maintenant:</span>
    </p>
    <div className="flex-1 flex-rows flex-center">
      <p>
        <ActivationButton className="pc-theme-gradient px18 pt16 pb18 fs20">
          <span>Activer votre pass Culture</span>
        </ActivationButton>
      </p>
      <p className="fs20 is-italic">
        <span>--</span>
        <span>&nbsp;ou&nbsp;</span>
        <span>--</span>
      </p>
      <p>
        <span className="fs20 is-bold">Explorez les offres</span>
      </p>
    </div>
  </div>
)

export default ActivationCard
