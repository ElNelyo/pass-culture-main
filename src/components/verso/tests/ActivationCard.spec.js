// jest --env=jsdom ./src/components/verso/tests/ActivationCard --watch
import React from 'react'
import { shallow } from 'enzyme'
import ActivationCard from '../ActivationCard'

describe('src | components | verso | ActivationCard', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = shallow(<ActivationCard />)
    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })
})
