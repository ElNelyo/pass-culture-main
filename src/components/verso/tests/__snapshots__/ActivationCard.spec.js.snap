// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | verso | ActivationCard should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivationCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="flex-0 text-center is-italic fs20"
        >
          <span
            className="is-block"
          >
            Et voilà !
          </span>
          <span
            className="is-block"
          >
            vous pouvez maintenant:
          </span>
        </p>,
        <div
          className="flex-1 flex-rows flex-center"
        >
          <p>
            <ActivationButton
              className="pc-theme-gradient px18 pt16 pb18 fs20"
              onClickHandler={[Function]}
              style={Object {}}
            >
              <span>
                Activer votre pass Culture
              </span>
            </ActivationButton>
          </p>
          <p
            className="fs20 is-italic"
          >
            <span>
              --
            </span>
            <span>
               ou 
            </span>
            <span>
              --
            </span>
          </p>
          <p>
            <span
              className="fs20 is-bold"
            >
              Explorez les offres
            </span>
          </p>
        </div>,
      ],
      "className": "flex-rows is-full-height is-white-text text-center",
      "id": "verso-tuto-activation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="is-block"
            >
              Et voilà !
            </span>,
            <span
              className="is-block"
            >
              vous pouvez maintenant:
            </span>,
          ],
          "className": "flex-0 text-center is-italic fs20",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Et voilà !",
              "className": "is-block",
            },
            "ref": null,
            "rendered": "Et voilà !",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "vous pouvez maintenant:",
              "className": "is-block",
            },
            "ref": null,
            "rendered": "vous pouvez maintenant:",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <ActivationButton
                className="pc-theme-gradient px18 pt16 pb18 fs20"
                onClickHandler={[Function]}
                style={Object {}}
              >
                <span>
                  Activer votre pass Culture
                </span>
              </ActivationButton>
            </p>,
            <p
              className="fs20 is-italic"
            >
              <span>
                --
              </span>
              <span>
                 ou 
              </span>
              <span>
                --
              </span>
            </p>,
            <p>
              <span
                className="fs20 is-bold"
              >
                Explorez les offres
              </span>
            </p>,
          ],
          "className": "flex-1 flex-rows flex-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ActivationButton
                className="pc-theme-gradient px18 pt16 pb18 fs20"
                onClickHandler={[Function]}
                style={Object {}}
              >
                <span>
                  Activer votre pass Culture
                </span>
              </ActivationButton>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <span>
                  Activer votre pass Culture
                </span>,
                "className": "pc-theme-gradient px18 pt16 pb18 fs20",
                "onClickHandler": [Function],
                "style": Object {},
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Activer votre pass Culture",
                },
                "ref": null,
                "rendered": "Activer votre pass Culture",
                "type": "span",
              },
              "type": [Function],
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  --
                </span>,
                <span>
                   ou 
                </span>,
                <span>
                  --
                </span>,
              ],
              "className": "fs20 is-italic",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "--",
                },
                "ref": null,
                "rendered": "--",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " ou ",
                },
                "ref": null,
                "rendered": " ou ",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "--",
                },
                "ref": null,
                "rendered": "--",
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="fs20 is-bold"
              >
                Explorez les offres
              </span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Explorez les offres",
                "className": "fs20 is-bold",
              },
              "ref": null,
              "rendered": "Explorez les offres",
              "type": "span",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="flex-0 text-center is-italic fs20"
          >
            <span
              className="is-block"
            >
              Et voilà !
            </span>
            <span
              className="is-block"
            >
              vous pouvez maintenant:
            </span>
          </p>,
          <div
            className="flex-1 flex-rows flex-center"
          >
            <p>
              <ActivationButton
                className="pc-theme-gradient px18 pt16 pb18 fs20"
                onClickHandler={[Function]}
                style={Object {}}
              >
                <span>
                  Activer votre pass Culture
                </span>
              </ActivationButton>
            </p>
            <p
              className="fs20 is-italic"
            >
              <span>
                --
              </span>
              <span>
                 ou 
              </span>
              <span>
                --
              </span>
            </p>
            <p>
              <span
                className="fs20 is-bold"
              >
                Explorez les offres
              </span>
            </p>
          </div>,
        ],
        "className": "flex-rows is-full-height is-white-text text-center",
        "id": "verso-tuto-activation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="is-block"
              >
                Et voilà !
              </span>,
              <span
                className="is-block"
              >
                vous pouvez maintenant:
              </span>,
            ],
            "className": "flex-0 text-center is-italic fs20",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Et voilà !",
                "className": "is-block",
              },
              "ref": null,
              "rendered": "Et voilà !",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "vous pouvez maintenant:",
                "className": "is-block",
              },
              "ref": null,
              "rendered": "vous pouvez maintenant:",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                <ActivationButton
                  className="pc-theme-gradient px18 pt16 pb18 fs20"
                  onClickHandler={[Function]}
                  style={Object {}}
                >
                  <span>
                    Activer votre pass Culture
                  </span>
                </ActivationButton>
              </p>,
              <p
                className="fs20 is-italic"
              >
                <span>
                  --
                </span>
                <span>
                   ou 
                </span>
                <span>
                  --
                </span>
              </p>,
              <p>
                <span
                  className="fs20 is-bold"
                >
                  Explorez les offres
                </span>
              </p>,
            ],
            "className": "flex-1 flex-rows flex-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ActivationButton
                  className="pc-theme-gradient px18 pt16 pb18 fs20"
                  onClickHandler={[Function]}
                  style={Object {}}
                >
                  <span>
                    Activer votre pass Culture
                  </span>
                </ActivationButton>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <span>
                    Activer votre pass Culture
                  </span>,
                  "className": "pc-theme-gradient px18 pt16 pb18 fs20",
                  "onClickHandler": [Function],
                  "style": Object {},
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Activer votre pass Culture",
                  },
                  "ref": null,
                  "rendered": "Activer votre pass Culture",
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    --
                  </span>,
                  <span>
                     ou 
                  </span>,
                  <span>
                    --
                  </span>,
                ],
                "className": "fs20 is-italic",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "--",
                  },
                  "ref": null,
                  "rendered": "--",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " ou ",
                  },
                  "ref": null,
                  "rendered": " ou ",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "--",
                  },
                  "ref": null,
                  "rendered": "--",
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="fs20 is-bold"
                >
                  Explorez les offres
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Explorez les offres",
                  "className": "fs20 is-bold",
                },
                "ref": null,
                "rendered": "Explorez les offres",
                "type": "span",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
